kernel/src/common/utils.rs

```diff
@@ -18,21 +15,15 @@
 /// destructor.
 #[macro_export]
 macro_rules! static_init {
-    ($T:ty, $e:expr) => {
-        // Ideally we could use mem::size_of<$T>, uninitialized or zerod here
-        // instead of having an `Option`, however that is not currently possible
-        // in Rust, so in some cases we're wasting up to a word.
-        {
-            use core::{mem, ptr};
-            // Statically allocate a read-write buffer for the value, write our
-            // initial value into it (without dropping the initial zeros) and
-            // return a reference to it.
-            static mut BUF: Option<$T> = None;
-            let tmp : &'static mut $T = mem::transmute(&mut BUF);
```

Comment:
I am not sure, but I think this is actually incorrect even now, i.e `Option<T>` could be layed out in memory beginning with the enum tag which this code ignores and uses for the value which could e.g. break alignment. It also creates the reference to a `T` before that `T` is initialized which still undecided but will possibly become undefined behavior.

When using `Option<T>`, I suspect this should have read something like
```
ptr::write(&mut BUF, Some($e));
let tmp: &'static mut $T = match &mut BUF {
  Some(buf) => buf,
  None => unreachable_unchecked(),
};
```